src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -351,6 +354,125 @@ private void decrypt(SymmetricKey encryptionKey) throws CouchbaseLiteException {
     }
 
     ///////////////////////////////////////////////////////////////////////////
+    // ENCRYPTABLE STORE
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public void setEncryptionKey(SymmetricKey key) {
+        encryptionKey = key;
+    }
+
+    @Override
+    public Action actionToChangeEncryptionKey(final SymmetricKey newKey) {
```

Comment:
The method is a part of the EncryptableStore interface so basically it comes together with setEncryptionKey() method. Currently the SQLiteStore itself also implements the encryption logic and it uses storageEngine.supportEncryption() method to determine if the encryption feature is available.

In my opinion: 
1. It's ok to be here unless we decide to extract the encryption feature out of a regular SQLiteStore (e.g. create a new SQLCipherStore). However making SQLCipherStore separately would need to have a factory object to determine what store to be created (which I think we already have that for creating SQLiteStore and ForestDBStore). 

2. If possible, let wait until the ForestDBStore encryption get implemented because they should get implemented in the same or similar structure. 

3. I don't like to say this but for code maintenance to match with iOS, sometimes it's useful to stay close to the iOS implementation as much as possible. Anyway, this implementation is currently matched with the iOS :)